### PR TITLE
[feat] Implement HomePage

### LINKED ISSUE #40: [feat] Implement HomePage
Homepage Screen Should Have Following Features:

1.  It Should be Responsive.
2. Users should be able to search on City.
3. Upon Search, it should show the user list of workspace along with filters
4.  If no workspace is present it should display a proper msg.

### PR DESCRIPTION
**Pull request by nitishmittal1990**
# Description

- [x] Included Router in App
- [x] Implement HomePage View
- [x] Implement Mock API Call to be integrated with Backend later
- [x] Navigate to WorkSpace Route on Success
- [x] Responsive

Closes #40 

### Desktop View: 
homepage

### Mobile View:
Screenshot 2020-02-18 at 6 20 23 PM

### CODE REVIEW

File: client/src/views/WorkSpaceList/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from "react";
+
+class WorkSpaceList extends Component {
```
Comment (nimish-gupta) [inline client/src/views/WorkSpaceList/index.js:3]: why class here?

File: client/src/views/Homepage/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+
+import style from "./Homepage.module.css";
+
+import { Grid, Button, TextField } from "../../components/Generic";
+import { getWorkSpaceData } from "../../apis/getWorkSpace";
+
+class Homepage extends Component {
```
Comment (nimish-gupta) [inline client/src/views/Homepage/index.js:8]: you can switch to functional components and use hooks over there.

File: client/src/views/Homepage/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        error: false,
+        searchTerm: value
+      }));
+    }
+  };
+
+  getWorkSpace = event => {
+    if (this.state.searchTerm.length > 2) {
+      event.target.setAttribute("disabled", true);
+      getWorkSpaceData().then(data => {
```
Comment (nimish-gupta) [inline client/src/views/Homepage/index.js:37]: please switch to `async/await`syntax, try to use es6 features.

**Review by nimish-gupta — APPROVED** [review]
Just resolve the comments and then LGTM.

File: client/src/views/WorkSpaceList/index.js (hunk 1 above)
Comment (nitishmittal1990) [reply client/src/views/WorkSpaceList/index.js:3]: ok will change to functional.

File: client/src/views/Homepage/index.js (hunk 2 above)
Comment (nitishmittal1990) [reply client/src/views/Homepage/index.js:8]: ok will change to functional.

**Review by vineetvk01 — APPROVED** [review]
LGTM After Nimish's changes

**Pull request merged**